torch/_dynamo/variables/builtin.py

```diff
@@ -515,46 +536,68 @@ def call_function(
+                            args[0].as_proxy().node.meta["example_value"]
+                            is ret.as_proxy().node.meta["example_value"]
+                        )
+                        ret = args[0]
+                    return ret
 
                 proxy = tx.output.create_proxy(
                     "call_function",
                     fn,
                     *proxy_args_kwargs(args, kwargs),
                 )
                 if any(isinstance(arg, FakeItemVariable) for arg in args):
-                    return wrap_fx_proxy_cls(
+                    ret = wrap_fx_proxy_cls(
                         FakeItemVariable,
                         tx,
                         proxy,
                     )
                 elif self.unspec_python_args(*args, **kwargs):
                     _args, _kwargs = self.unwrap_unspec_args_kwargs(args, kwargs)
-                    raw_value = self.fn(*_args, **_kwargs)
+                    raw_value = fn(*_args, **_kwargs)
 
                     need_unwrap = any(
                         x.need_unwrap
                         for x in itertools.chain(args, kwargs.values())
                         if isinstance(x, variables.UnspecializedPythonVariable)
                     )
 
-                    return wrap_fx_proxy_cls(
+                    ret = wrap_fx_proxy_cls(
                         UnspecializedPythonVariable,
                         tx,
                         proxy,
                         raw_value=raw_value,
                         need_unwrap=need_unwrap,
                     )
                 elif all(isinstance(x, SymNodeVariable) for x in args):
-                    return SymNodeVariable.create(tx, proxy, None)
+                    ret = SymNodeVariable.create(tx, proxy, None)
                 else:
                     # Work around for vision_maskrcnn due to precision difference
                     # specialize the dividend when float divide by tensor
-                    if self.fn is operator.truediv and isinstance(
+                    if fn is operator.truediv and isinstance(
                         args[0], variables.UnspecializedPythonVariable
                     ):
                         args[0] = args[0].convert_to_constant(tx)
-                    return wrap_fx_proxy(tx, proxy)
+                    ret = wrap_fx_proxy(tx, proxy)
+
+                # TODO: does this also apply to SymNodeVariable? A: Yes if:
+                # 1. (legal) SymNodeVariable mutations are propagated via `example_value`
+                # 2. (required) SymNodeVariable can have attribute mutations
+                if fn in self._self_assigning_ops() and isinstance(
+                    args[0], variables.TensorVariable
+                ):
+                    assert (
+                        args[0].as_proxy().node.meta["example_value"]
+                        is ret.as_proxy().node.meta["example_value"]
+                    )
+                    ret = args[0]
```

Comment:
It feels like it would be cleaner if we didn't create a fresh TensorVariable only to get rid of it because it was bad. Maybe we can do something about this in `wrap_fx_proxy`? E.g., what if that was responsible for detecting that an output aliased, and then reusing the old variable?

Reply:
Hmm. The problem is that making wrap_fx_proxy aware of in-place ops/aliasing in the fx graph might be a little annoying. Today, it simply blindly wraps stuff.

We could do memoization by storing a cache of all the TensorVariables it has ever produced under a given fake_mode? ...

Reply:
Option 1 seems best, but do you think it is worth doing?

Reply:
Actually, probably 2. is the best as the lifetimes of output graph and VTs are aligned.

Not too sure if there would be an issue for higher order ops.

Reply:
You only need to do (3). Operators are "functional": they won't manufacture references to other Tensors out of no where (well, OK, AMP caching breaks that, but AMP caching is cancer anyway). So for VT aliasing, checking the inputs is precisely sufficient.

Reply:
Blah, you're right, `wrap_fx_proxy` doesn't take the VTs. OK, fine, let's do it this way lol